lib/ansible/modules/source_control/git_config.py

```diff
@@ -188,6 +208,11 @@ def main():
     else:
         new_value = None
 
+    if params['state'] == "absent":
+        config_state = params['state']
```

Comment:
The name of this variable implies that it could also contain `"present"`, but it doesn't currently do that. I feel like it could be made clearer.

Reply:
I originally conceptualized something like this:
```python
if params['state'] == "absent":
    config_state = "--unset"
else:
    config_state = None
```
...since the default behavior of the `git_config` module is to set `name` to `value` without the need for `state: present`. If `config_state` turns out to be `present` then there is no change to the flags passed to `git config`. But if `config_state` is not None we know the user wants to `--unset`.

I'm all for more clarity though being new to the code I appreciate input on the best path forward.

This seems to offer less insight
```python
if params['state']:
    config_state = params['state']
```

This seems repetitive...
```python
if params['state'] == "absent":
    config_state = params['state']
elif params['state'] == "present":
    config_state = params['state']
else:
    config_state = None
```

Reply:
Maybe this?

    if params['state']:
        config_state = params['state']
    else:
        config_state = None

The AnsibleModule class has already verified that the value in params['state'] (if specified) is one of 'present' or 'absent', so you don't need to check that again yourself.